### ISSUE TITLE
Tackle animation in PF4

### ISSUE THREAD

**Issue by matthewcarleton** [patternfly/patternfly-next#938]
We need to officially kick off the conversation around animation in PF4. For reference we can look at 
https://material.io/design/motion/
https://www.lightningdesignsystem.com/guidelines/animation/styleguidelines/
I think our first step is determine our overall approach to animation. 

@kybaker @mceledonia Let's meet and kick this off. 

We can also cover the work that needs to happen on the Openshift POC side.

**Comment by mcoker** [comment]
FWIW I was looking at the [redhat-theme repo](https://gitlab.corp.redhat.com/uxdd/redhat-theme) and they have a [global animation var](https://gitlab.corp.redhat.com/uxdd/redhat-theme/blob/dev/src/maps/_general.scss#L22-24) defined as `.2s ease-in`.

**Comment by matthewcarleton** [comment]
ya, I think that's a good call. It feels like we need to do a sweep of all the components and determine where animations belong. Having something like this defined will be helpful so we have a consistent approach. Although, I wonder if we need more than just one. Depending on the animation it could need something longer/shorter.

**Comment by mceledonia** [comment]
Definitely agreed here, let's start with the one and see how it works across the system. If we need more we can always add them, but it won't hurt to start simple. My overall approach to animation was very short, fast animations with easing, so its possible that one global var might fit that style.

**Comment by jgiardino** [comment]
During the CSS meeting today, we were reviewing the Alert and https://github.com/patternfly/patternfly-next/pull/1726, and the subject of animation came up. To make the toast variant accessible to screen readers, we are using css animation to load the wrapper .2s  before the contents (which is required for JAWS and NVDA). As part of this discussion, we talked about what to do with the wrapper while we wait for the contents to load. 

I'd like to nominate the Alert as the first component that we work on for animation. It definitely fits the use case as a component that benefits from animation, but also, having this css animation solution for accessible toasts is odd without a fully designed animation.

**Comment by mcarrano** [comment]
Clearing assignees until someone can actively pick this one up. @rachael-phillips @LHinson did not see this on our roadmap so added to the "unplanned" column. We should discuss when this can get done as I believe it can significantly enhance the PF4 experience.

**Comment by jgiardino** [comment]
As part of animation, we should do some research on how to utilize the `prefers-reduced-motion` media query, similar to a suggestion noted in https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior#Accessibility_concerns

**Comment by matthewcarleton** [comment]
@mcarrano curious why this got closed

**Comment by mcarrano** [comment]
I believe that @LHinson and I made a decision to close it as it was something we felt we were unable to take action on in the near term. I am in favor of introducing animation to the library, but we need to better identify what that would look like and have a plan to introduce this. @matthewcarleton @mceledonia @mcoker is this something we should discuss further?

**Comment by mcoker** [comment]
@mcarrano I vote yes!

**Comment by matthewcarleton** [comment]
Ya - I think this has come up recently in a couple different places. As PF has become solid and more widely used the lack of animation has become a bit more obvious. It's a good time to talk about it and determine the added aesthetic/UX/A11y benefits. We do have an animation working group that we kicked off as part of the OCP breakaway we did recently - I think this could help with this effort. We definitely want to align with PF in that group.
